tests/Functional/Net/SFTPUserStoryTest.php

```diff
@@ -649,5 +649,22 @@ public function testStatVsLstat($sftp)
         $this->assertSame($stat['type'], NET_SFTP_TYPE_SYMLINK);
 
         $sftp->enableStatCache();
+
+        return $sftp;
+    }
+
+    /**
+     * @depends testStatVsLstat
+     * @group github830
+     */
+    public function testEndlessLoopOnUpload($sftp)
```

Comment:
I can do that...

Although tbh I think the code ought to be updated such that the "second hunk" isn't even possible. Quoting from the code comments:

```php
     * $start and $local_start give you more fine grained control over this process and take precident over self::RESUME
     * when they're non-negative. ie. $start could let you write at the end of a file (like self::RESUME) or in the middle
     * of one. $local_start could let you start your reading from the end of a file (like self::RESUME_START) or in the
     * middle of one.
     *
     * Setting $local_start to > 0 or $mode | self::RESUME_START doesn't do anything unless $mode | self::SOURCE_LOCAL_FILE.
```

Based on that it doesn't seem like it should be doing `fseek($fp, $offset)`, since `$offset` is based on `$start`. ie. I think they ought to be completely disjoint.

Here's the commit that added it:

https://github.com/phpseclib/phpseclib/commit/cbe156f0cbc7e715731b841482a1b1625df31250

Truth by told, I don't really like how the `put` function has so many different parameters and it's something I'd like to change, as elaborated upon here:

https://github.com/phpseclib/phpseclib/pull/655#issuecomment-92600348